cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -732,8 +732,18 @@ private V1ObjectMeta createDeployment(V1ObjectMeta metadata,
 
     V1Deployment deployment = buildDeployment(metadata, runtimeSpecs, runtimeConfigLocation);
 
-    deployment = appsApi.createNamespacedDeployment(programRuntimeNamespace, deployment, "true", null, null);
-    LOG.info("Created Deployment {} in Kubernetes", metadata.getName());
+    try {
+      deployment = appsApi.createNamespacedDeployment(programRuntimeNamespace, deployment, "true", null, null);
+      LOG.info("Created Deployment {} in Kubernetes", metadata.getName());
+    } catch (ApiException e) {
+      if (e.getCode() == HttpURLConnection.HTTP_CONFLICT) {
```

Comment:
we don't expect this to ever happen right?

Reply:
AFAIK it should basically never happen unless the launching job fails after the deployment is created but before the state of the program can be updated.